web/app/routes/job.js

```diff
@@ -15,19 +15,80 @@
  */
 
 import Ember from 'ember';
+import Scheduler from 'dr-elephant/utils/scheduler';
 
 export default Ember.Route.extend({
   ajax: Ember.inject.service(),
-  beforeModel: function (transition) {
+  beforeModel: function(transition) {
+    let loginController = this.controllerFor('login');
+    loginController.set('previousTransition', transition);
+>>>>>>> 60b0f9c... Implementation of Authentication and Authorization for TuneIn purpose
     this.jobid = transition.queryParams.jobid;
   },
 
-  model(){
+  model() {
     return Ember.RSVP.hash({
-      jobs:   this.store.queryRecord('job', {jobid: this.get("jobid")}),
-      tunein: this.store.queryRecord('tunein', {id: this.get("jobid")})
+      jobs: this.store.queryRecord('job', {
+        jobid: this.get("jobid")
+      }),
+      tunein: this.store.queryRecord('tunein', {
+        id: this.get("jobid")
+      })
     });
   },
+  doLogin(schedulerUrl, cluster) {
+    //confirm if the user want to proceed with login
+    var userWantToLogin = confirm("To perform this action user needs to login. Are you sure to proceed?")
+    if (userWantToLogin) {
+      this.transitionTo('login').then((loginRoute) => {
+        loginRoute.controller.set('schedulerUrl', schedulerUrl);
+        loginRoute.controller.set('cluster', cluster);
+      });
+    }
+  },
+  getUserAuthorizationStatus(jobdefid, schedulerUrl, cookieName) {
+    var authorizationStatus
+    const is_authorised_key = "hasWritePermission"
+    $.ajax({
+      url: "/rest/userAuthorization",
+      type: "GET",
+      data: {
+        sessionId: Cookies.get(cookieName),
+        jobDefId: jobdefid,
+        schedulerUrl: schedulerUrl
+      },
+      async: false
+    }).then((response) => {
+      if (response.hasOwnProperty("hasWritePermission") && response.hasWritePermission === "true") {
+      if (response.hasWritePermission === "true") {
+        authorizationStatus = "authorised";
+      } else {
+        authorizationStatus = "unauthorised";
+      }
+    } else if (response.hasOwnProperty("error")) {
+      if (response.error === "session") {
+        console.log("Previous session_id expired, so proceed to login")
+        authorizationStatus = "session_expired";
+      } else {
+        //Some other error occurred
+        authorizationStatus = "error"
+      }
+    }
+  },
+    (error) => {
+      switch (error.status) {
+        case 400:
+          alert(error.responseText);
```

Comment:
nit: avoid blocking UI's. Prefer inline error messages. 
Applicable across the file

Reply:
The alerts have been now changed to Notifications.